pkg/operatormanager/operatormanager.go

```diff
@@ -353,6 +362,12 @@ func (m *OperatorManager) createNewClientObject(ctx context.Context, obj client.
 		}
 	case *appsv1.Deployment:
 		m.log.Info("handling Deployment")
+		if len(v.Spec.Template.Spec.Containers) != 1 {
+			m.log.Info("Unexpected number of containers in deployment, ignoring.")
+		} else if m.options.OperatorImage != "" {
+			m.log.Info("Patching operator image", "image", m.options.OperatorImage)
+			v.Spec.Template.Spec.Containers[0].Image = m.options.OperatorImage
```

Comment:
Is this kind of a new feature? What does this do?

Reply:
Well... in this code, we parse the [YAML file with the complete operator deployment](https://github.com/fi-ts/postgreslet/blob/main/external/svc-postgres-operator.yaml) and modify it on the fly. This eases the maintenance of the deployment (we only need to update one file) and also makes our changes more visible.

In this specific case, I overwrite the [image](https://github.com/fi-ts/postgreslet/blob/main/external/svc-postgres-operator.yaml#L326) value in the deployment with the one configured in the postgreslet. This is a new feature. There is only one deployment contained in the YAML currently, so no special care is taken for finding that deployment, but the number of containers might vary in the future, so there is a little bit of a check in here.